core/libtexpdf-output.lua

```diff
@@ -117,30 +128,42 @@ SILE.outputters.libtexpdf = {
     -- position (cursorX + width - remember that the box's "width"
     -- is actually the shaped x_advance).
     if value.complex then
-      for i = 1, #(value.items) do
-        local glyph = value.items[i].gid
-        local buf = string.char(math.floor(glyph % 2^32 / 2^8)) .. string.char(glyph % 0x100)
-        pdf.setstring(cursorX + (value.items[i].x_offset or 0), cursorY + (value.items[i].y_offset or 0), buf, string.len(buf), font, value.items[i].glyphAdvance)
-        cursorX = cursorX + value.items[i].width
+      for i = 1, #value.items do
+        local buf = glyph2string(value.items[i].gid)
+        self:setCursor(value.items[i].x_offset or 0, value.items[i].y_offset or 0, true)
+        self:_drawString(buf, value.items[i].glyphAdvance)
+        self:setCursor(value.items[i].width, 0, true)
```

Comment:
@simoncozens Off the top of your head can you tell me anything about what these 3 lines were supposed to do? I ported them across with the math as-is, but I'm concerned because I don't get what it does and substituting this doesn't break any of our tests:

```suggestion
        self:_drawString(buf, 0)
```

Reply:
Nevermind, I finally found a case where this matters and will add a test for it so this code path actually gets covered.